### ISSUE TITLE
grid design with imagesLoaded, large image-based dataset, and react

### ISSUE THREAD

**Issue by dylanfpaul** [desandro/masonry#901]
Forgive the absence of a test case but I have a question about component design relating to masonry, a large image-based dataset, and react. My first question is if whether or not the 'appended' method is appropriate for adding items to the grid when using React. I thought of this question when I tried playing around with appending items to a grid in React only to realize that React's render method kind of covers this. If React's render method can fulfil the requirement of adding items to the grid (in combination with imagesLoaded), how should I design my grid component(s) to append a set number of items per iteration _without having imagesLoaded re scan all of the items in the grid_ (the ones that have already been loaded). I am asking this because I plan to have a large image-based data set and I am looking for a way to prevent performance issues when using imagesLoaded (especially when the data set has reached a large number of items).

My initial thoughts about my concerns were to split my masonry grid into many smaller divisions - Is this possible with masonry's current functionality? Can I have one large grid with many containers containing grid items? I thought this could possibly resolve the imagesLoaded re-scanning the whole grid every time issue by only re-scanning the most recently created division on the end of the grid.

**Comment by desandro** [comment]
> whether or not the 'appended' method is appropriate for adding items to the grid when using React.

Perhaps. I do not have any expertise with React, so I cannot say. I'm interested to hear your experience.

**Comment by dylanfpaul** [comment]
Using React, I actually split the grid up into multiple rows, (each row
being a numbered div, but all of them still within one masonry grid
container). Each div representing a new row then gets rendered to the DOM
and has imagesLoaded called upon it. This means that imagesLoaded no longer
scans the whole dataset everytime something new is added, it only scans the
incoming row. I also did not use any masonry methods to append new items
into the grid, I simply just rendered them using React life-cycle logic and
React's render() method. This is definitely not the intended way to use
masonry and imagesLoaded but it worked for my use-case. An important thing
to note is that this workaround does not work at all with isotope.
